contracts/optimize/SlotHelper.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+library SlotHelper {
+    uint256 internal constant SLOTDATA_RIGHT_SHIFT = 32;
+    uint256 internal constant LEN_OFFSET = 224;
+    uint256 internal constant FIRST_SLOT_DATA_SIZE = 28;
+
+    function putRaw(
+        mapping(uint256 => bytes32) storage slots,
+        bytes memory datas
+    ) internal returns (bytes32 mdata) {
+        // warn: if data as ptr to move,your should keep "data.length" by another ptr
```

Comment:
the comment can be improved: combining the words after "if" is not a complete sentence.  Further, I am not quite sure the comment means.